### ISSUE TITLE
2.0: crash - Database is not open

### ISSUE THREAD

**Issue by Codeido** [couchbase/couchbase-lite-ios#1942]
Hi guys,

I see that sometimes the live query is not registered because the database is not opened, and i have a crash when i try to recreate the database in this way:

```
try database?.delete()
database = try Database(name: "my-db")
liveQuery1()
liveQuery2()
```
the crash is this:
`*** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'Database is not open.'`

what do you think? Is a bug?

--------
* Version:   2.0 B16
* Client OS: iOS 11
* Server:   -

**Comment by pasin** [comment]
This is strange. If database cannot be open `try Database(name: "my-db")` should throw an error. This is probably a bug.

**Comment by Codeido** [comment]
If i call this before the database?.delete() method the problem is fixed. Do you know why?
```
query1?.stop()
query2?.stop()
```

**Comment by pasin** [comment]
I think it's because the live query which uses the deleted database is not stopped yet when the database is deleted. Manually stopping the live query is the right workaround. CBL should also stop live query gracefully when the database is closed or deleted so this needs to be fixed.

**Comment by rajagp** [comment]
Tracked in https://github.com/couchbase/couchbase-lite-ios/issues/1958

**Comment by pasin** [comment]
As mention in #1958, the feature that stops the live query when database is deleted has been implemented and merged into feature/2.0 branch. We can close this issue now.
